core/src/solution_submission.rs

```diff
@@ -229,6 +232,24 @@ async fn retry_with_gas_price_increase(
         )
     };
 
+    let wait_for_deadline = || async {
+        match BatchId::from(batch_index)
+            .solve_end_time()
+            .duration_since(SystemTime::now())
+        {
+            // Add some extra time because of desync between real current time and ethereum current
+            // block time.
+            Ok(remaining) => async_std::task::sleep(remaining + Duration::from_secs(30)).await,
+            Err(err) => {
+                log::warn!(
+                    "could not calculate remaining time to solve batch: {:?}",
+                    err
+                );
+                future::pending().await
```

Comment:
Should this rather resolve immediately? Could this happen if we increased the gas limit with 15s remaining and so the next time we try to compute the remaining time it is negative?

Reply:
Good point.